### ISSUE TITLE
Note wrongly added because snapshot's too old

### ISSUE THREAD

**Issue by bmallet** [organicmaps/organicmaps#8463]
Someone contrbuted to OSM with a note (https://www.openstreetmap.org/note/4283551) saying the shop is closed, nice...   but this shop has been deleted from OSM one year ago :-(
This contributor has downloaded his map in 2022-07-18 if i do understand the trouble.

I think, before sending a correction or a note to OSM, Organic Maps should check that the downloaded map is not too old, note more than two months, for example. If it's not the case, you should prevent the contribution (we do have enough notes in OSM;-).

**Comment by bmallet** [comment]
Ah and by the way, thank you for this nice app which is Organic Maps, this kind of app is a good way for the people to discover OSM.

**Comment by Jean-BaptisteC** [comment]
Normally if maps is too old, its not possible to edit map

**Comment by patepelo** [comment]
related to #6936 and #7930

**Comment by biodranik** [comment]
Map data edits are blocked without updating maps. Posting notes is not blocked (why should it? Useful notes can come about different objects, including missing ones).

**Comment by bmallet** [comment]
OK but to get useful and right comment in a note it's better if this note is written after have looked the actual map.
Case i gave in example (https://www.openstreetmap.org/note/4283551) : the organic maps contributor says a shop is closed because he has seen it opened on the map but in fact this shop has been closed in OSM on 2023-07-05 (https://www.openstreetmap.org/node/3501719732/history)
Result : a note useless
And in your example "Useful notes can come about different objects, including missing ones", same kind of trouble could occur : the organic maps contributor write a note for a "new" shop he sees but, because his map is 6 months old, this shop already exit, in fact, in the OSM database for 3 months : definitely this is not an "useful" note.
I do think that telling people who want to contribute, even with notes : "sorry, your map is too old (2 months old) ; before posting a note, please upload an actual version" would be a better way.

**Comment by biodranik** [comment]
Right, this issue should be solved by promoting/forcing users to update maps. It can be done by displaying the "Update Map" button in the Place Page, and by displaying a warning/note when editing a note.

@euf @kirylkaveryn @Jean-BaptisteC  is it hard to add an "Update Map" button into the Place Page? Do you have any ideas on where to put it? The first "Download" button is already displayed in PP when an empty map region is selected. Maybe always displaying update there will be annoying enough to promote map update? Then just reusing the same button/code but only use a different translation for it would be sufficient?

**Comment by kirylkaveryn** [comment]
> Do you have any ideas on where to put it?

Maybe we can reuse the `edit place` button in pp?
With text "Update map to edit place".

![image](https://github.com/organicmaps/organicmaps/assets/79797627/05a5b654-f793-49c5-b27b-eb1984258ca7)

And in case of Add Place we can show the alert when user try to open the "add place" with buttons: "update map" (that will opens the download screen) and "cancel" the just close an alert.

**Comment by biodranik** [comment]
This is a cool idea, is it possible to display the download indicator (simplified, just text percents) in the same Edit Place button?

Regarding Add Place, we can offer an action with two items "Update <region> now" and "Update later", WDYT?

**Comment by kirylkaveryn** [comment]
@biodranik can you please check this out?
https://www.figma.com/design/zxnZsa3SBaSAPWG0BmZ3Xh/OM-UpdateMap?node-id=0-1&t=hzbrTvBTUmRC6Jam-1

**Comment by kirylkaveryn** [comment]
> Regarding Add Place, we can offer an action with two items "Update now" and "Update later", WDYT?

In case of "Update later" the alert will be closed without any actions right?

**Comment by bmallet** [comment]
i did not write to "LAPR_luis" and looking at his profile (https://www.openstreetmap.org/user/LAPR_luis) it looks like he's not a frequent contributor
